### PR TITLE
Added support for new version of docgen and add support for implements

### PR DESCRIPTION
**Pull request by NbOpposite**
Adds support for [this PR](https://github.com/discordjs/docgen/pull/3), backward compatible with version docgen:0.8.0 format.

We've chatted on discord, do you really need a more detailed description of the PR?

### CODE REVIEW

File: src/DocElement.js [hunk 1]
```diff
@@ -64,8 +65,27 @@ class DocElement extends DocBase {
...
-      const baseClass = this.doc.get(this.extends[0]) || this.extends[0]
-      name += ` (extends **${typeof baseClass === 'string' ? baseClass : baseClass.link}**)`
+      if(typeof this.extends[0] === 'string') {
+        // docgen 0.8.0 format
+        const baseClasses = this.extends.map(baseClass => (this.doc.get(baseClass) || { link: baseClass }).link)
+        name += ` (extends **${ baseClasses.join('** and **')}**)`
+      } else {
+        // docgen 1.0.0 format
+        const baseClasses = this.extends.map(baseClass => flatten(baseClass).map(el => (this.doc.get(el) || { link: el }).link))
+        name += ` (extends **${ baseClasses.map(e=>e.join('')).join('** and **')}**)`
```
Comment (TeeSeal) [inline src/DocElement.js]: Spaces around `=>` please

(same below)

**Comment by TeeSeal** [comment]
Also, if possible, try to keep line length under 100 characters.
Don't hesitate to chain methods to new lines:
```js
thing
  .doStuff()
  .doOtherStuff()
```
or create variables only for the sake of readability.

**Comment by NbOpposite** [comment]
(Now the longest line is 99 characters 😂)

**Comment by NbOpposite** [comment]
Oh, I missed the first comment, I'm on refactoring it out to separate functions

**Pull request merged**